modules/sfp_darksearch.py

```diff
@@ -0,0 +1,126 @@
+# -*- coding: utf-8 -*-
+# -------------------------------------------------------------------------------
+# Name:         sfp_darksearch
+# Purpose:      Searches the Darksearch.io Tor search engine for content related 
+#               to the domain in question.
+#
+# Author:      <bcoles[at]gmail[.]com>
+#
+# Created:     2019-05-11
+# Copyright:   (c) bcoles 2019
+# Licence:     GPL
+# -------------------------------------------------------------------------------
+
+import json
+import time
+import urllib
+from sflib import SpiderFoot, SpiderFootPlugin, SpiderFootEvent
+
+class sfp_darksearch(SpiderFootPlugin):
+    """Darksearch:Footprint,Investigate:Search Engines::Search the Darksearch.io Tor search engine for mentions of the target domain."""
+
+    # Default options
+    opts = {
+        'max_pages': 20
+    }
+
+    # Option descriptions
+    optdescs = {
+        'max_pages': "Maximum number of pages of results to fetch."
+    }
+
+    results = dict()
+    errorState = False
+
+    def setup(self, sfc, userOpts=dict()):
+        self.sf = sfc
+        self.__dataSource__ = "Darksearch"
+        self.results = dict()
+        self.errorState = False
+
+        for opt in userOpts.keys():
+            self.opts[opt] = userOpts[opt]
+
+    def watchedEvents(self):
+        return ['DOMAIN_NAME', 'EMAILADDR']
+
+    def producedEvents(self):
+        return ['DARKNET_MENTION_URL', 'DARKNET_MENTION_CONTENT', 'SEARCH_ENGINE_WEB_CONTENT']
+
+    # https://darksearch.io/apidoc
+    def query(self, qry, page):
+        params = {
+            'query': qry.encode('raw_unicode_escape'),
+            'page': str(page)
+        }
+
+        res = self.sf.fetchUrl("https://darksearch.io/api/search?" + urllib.urlencode(params),
```

Comment:
`urllib.urlencode()` is not Unicode safe afaik.

Reply:
What's best practice here? The framework appears to take a few approaches.

This approach is copypasta from another module (`sfp_openstreetmap`), in which you added support for unicode in a413501e78e6d295d8bee546bed6201511a7061b by using `encode('raw_unicode_escape')`.